coldfront/plugins/qumulo/utils/eib_billing.py

```diff
@@ -0,0 +1,187 @@
+    LEFT JOIN (SELECT aa.allocation_id, aa.value cost_center FROM allocation_allocationattribute aa JOIN allocation_allocationattributetype aat ON aa.allocation_attribute_type_id=aat.id WHERE aat.name='cost_center') AS cost_center ON a.id=cost_center.allocation_id
+    LEFT JOIN (SELECT aa.allocation_id, aa.value department_number FROM allocation_allocationattribute aa JOIN allocation_allocationattributetype aat ON aa.allocation_attribute_type_id=aat.id WHERE aat.name='department_number') AS department_number ON a.id=department_number.allocation_id
+    LEFT JOIN (SELECT aa.allocation_id, aa.value service_rate_category FROM allocation_allocationattribute aa JOIN allocation_allocationattributetype aat ON aa.allocation_attribute_type_id=aat.id WHERE aat.name='service_rate') AS service_rate_category ON a.id=service_rate_category.allocation_id
+    LEFT JOIN (SELECT aa.allocation_id, aa.id, aa.value storage_quota FROM allocation_allocationattribute aa JOIN allocation_allocationattributetype aat ON aa.allocation_attribute_type_id=aat.id WHERE aat.name='storage_quota') AS storage_quota ON a.id=storage_quota.allocation_id
+    JOIN (
+      SELECT allocation_attribute_id,
+        MAX(modified) recent_modified,
+        value
+      FROM allocation_historicalallocationattributeusage
+      GROUP BY allocation_attribute_id, DATE(modified)
+    ) AS most_recent_haau
+      ON storage_quota.id=most_recent_haau.allocation_attribute_id
+    JOIN allocation_allocation_resources ar ON ar.allocation_id=a.id
+    JOIN resource_resource r ON r.id=ar.resource_id
+    WHERE
+      r.name = 'Storage2'
+    AND
+      DATE(most_recent_haau.recent_modified) = '%s'
+    AND
+      astatus.name = 'Active'
+  ) AS data
+) AS report where report.billing_unit > 0;
+"""
+
+filename = ""
+
+def reset_filename() -> None:
+    global filename
+    filename = "/tmp/RIS-%s-storage2-active-billing.csv"
+
+def get_filename() -> str:
+    return filename
+
+
+def get_report_header() -> str:
+    return REPORT_HEADER
+
+
+def get_monthly_billing_query_template() -> str:
+    return QUERY_MONTHLY_BILLING
+
+
+def generate_monthly_billing_report(usage_date=datetime.today().replace(day=1).strftime(YYYY_MM_DD)) -> bool:
+    # The date when the billing report was generated
+    document_date = datetime.today().strftime("%m/%d/%Y")
+    # The first day of the service month
+    delivery_date = (datetime.strptime(usage_date, YYYY_MM_DD).replace(day=1) - timedelta(1)).replace(day=1).strftime(YYYY_MM_DD)
+    # The service month for billing
+    billing_month = datetime.strptime(delivery_date, YYYY_MM_DD).strftime("%B")
+
+    reset_filename()
+    global filename
+    # The temporary report file for the service month
+    filename = get_filename() % billing_month
+
+    monthly_billing_query = get_monthly_billing_query_template() % (document_date, billing_month, delivery_date, usage_date)
+    # logger.debug("Monthly billing query: %s", monthly_billing_query)
```

Comment:
I intentionally leave it for future reference in case for troubleshooting. I think it will be helpful for others, too before refactoring using ORM. But I can remove it if you don't feel the same.

Reply:
I won't demand it's removal.  It's a fair argument.

I can't guarantee that I wouldn't take it out in the future if I'm working on the file.  Not out of spite, but out of absentmindedness